frontend/discourse/app/components/d-otp/slot.gjs

```diff
@@ -0,0 +1,44 @@
+import Component from "@glimmer/component";
+import { isBlank } from "@ember/utils";
+import concatClass from "discourse/helpers/concat-class";
+
+const PLACEHOLDER_CHAR = "-";
+
+export default class Slot extends Component {
+  get showCursor() {
+    return this.args.isFocused && isBlank(this.args.char);
+  }
+
+  get displayChar() {
+    if (!isBlank(this.args.char)) {
+      return this.args.char;
+    }
+
+    if (this.showCursor) {
+      return "";
+    }
+
+    return PLACEHOLDER_CHAR;
+  }
+
+  get isPlaceholder() {
+    return this.displayChar === PLACEHOLDER_CHAR;
+  }
+
+  <template>
+    <div
+      class={{concatClass
+        "d-otp-slot"
+        (if @isFocused "--is-focused")
+        (if this.showCursor "--show-cursor")
+        (if this.isPlaceholder "--placeholder")
+      }}
+      data-index={{@index}}
+    >
+      <span aria-hidden="true">
+        &#8203;
+      </span>
```

Comment:
Ensure consistent size even when no text inside, as we use padding for sizing and want this to have the same height as our other components (inputs, buttons..)